x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/phases/shared/data_tier_allocation_field.tsx

```diff
@@ -46,35 +48,56 @@ export const DataTierAllocationField: FunctionComponent<Props> = ({
   isShowingErrors,
   errors,
 }) => {
+  const {
+    services: { cloud },
+  } = useKibana();
+
   return (
     <NodesDataProvider>
-      {(nodesData) => {
-        const hasNodeAttrs = Boolean(Object.keys(nodesData.nodesByAttributes ?? {}).length);
-
-        const renderDefaultAllocationNotice = () => {
-          if (phaseData.dataTierAllocationType !== 'default') {
-            return null;
-          }
+      {({ nodesByRoles, nodesByAttributes, isUsingLegacyDataRoleConfig }) => {
+        const hasNodeAttrs = Boolean(Object.keys(nodesByAttributes ?? {}).length);
 
-          const allocationNodeRole = getAvailableNodeRoleForPhase(phase, nodesData.nodesByRoles);
-          if (
-            allocationNodeRole !== 'none' &&
-            isNodeRoleFirstPreference(phase, allocationNodeRole)
-          ) {
-            return null;
-          }
+        const renderNotice = () => {
+          if (phaseData.dataTierAllocationType === 'default') {
+            const isCloudEnabled = cloud?.isCloudEnabled ?? false;
+            if (
+              isCloudEnabled &&
+              !isUsingLegacyDataRoleConfig &&
```

Comment:
This might sound silly, but can we extract this into a variable that offers some more meaning in the context of Cloud? For example, my understanding is that Cloud has two states: node roles are enabled or node roles are disabled. Reading a bit further, I think it'd be easier for me to wrap my head around this complex condition by doing the same thing with the other conditions. So maybe something like this would communicate the intention more clearly:

```js
              const isCloudEnabled = cloud?.isCloudEnabled ?? false;
              const isColdPhase = phase === 'cold';
              const areNodeRolesEnabledOnCloud = !isUsingLegacyDataRoleConfig;
              const hasColdNodeRoles = !nodesByRoles.data_cold?.length;
              
              if (
                isCloudEnabled &&
                isColdPhase &&
                areNodeRolesEnabledOnCloud &&
                hasColdNodeRoles
              ) {
                // Tell cloud users they can deploy cold tier nodes.
                return (
                  <>
                    <EuiSpacer size="s" />
                    <CloudDataTierCallout />
                  </>
                );
              }
```